src/Server/HTTPHandler.cpp

```diff
@@ -929,15 +929,25 @@ void HTTPHandler::handleRequest(HTTPServerRequest & request, HTTPServerResponse
             request_credentials.reset(); // ...so that the next requests on the connection have to always start afresh in case of exceptions.
         });
 
-        tryLogCurrentException(log);
+        /// Check if exception was thrown in used_output.finalize().
+        /// In this case used_output can be in invalid state and we
+        /// cannot write in it anymore. So, just log this exception.
+        if (used_output.isFinalized())
+        {
+            tryLogCurrentException(log, "Cannot flush data to client");
```

Comment:
tryLogCurrentException and maybe immediately return from handler? The code will be slightly more readable